index/scorch/scorch.go

```diff
@@ -438,11 +439,24 @@ func (s *Scorch) Batch(batch *index.Batch) (err error) {
 				segB.BytesWritten())
 		}
 		atomic.AddUint64(&s.iStats.newSegBufBytesAdded, bufBytes)
+
+		stats["num_vectors"] = make(map[string]int)
+		for _, result := range analysisResults {
+			result.VisitFields(func(field index.Field) {
+				if string(field.EncodedFieldType()) == "v" {
+					if _, exists := stats["num_vectors"][field.Name()]; !exists {
+						stats["num_vectors"][field.Name()] = 0
+					}
+
+					stats["num_vectors"][field.Name()] += 1
```

Comment:
does this include the nested vectors as well? like i see you're increasing the value by 1 rather than say length/dims over here.

Reply:
I'd discussed this with @Likith101 earlier and IIRC he said yes.

Reply:
If there's way we can avoid to individually increment the count for the metric and instead do it once - that _must_ be preferred.

Reply:
Updated this so that the flow goes to the zap layer to populate the stats.